parser/src/aspects/literal.rs

```diff
@@ -206,55 +214,52 @@ impl<'a> LiteralAspect<'a> for Parser<'a> {
         let token = self.cursor.next()?;
         match token.token_type {
             TokenType::IntLiteral => Ok(LiteralValue::Int(token.value.parse::<i64>().map_err(
-                |e| {
-                    match e.kind() {
-                        IntErrorKind::PosOverflow | IntErrorKind::NegOverflow => self
-                            .expected::<()>("Integer constant is too large.")
-                            .unwrap_err(),
-                        _ => self.mk_parse_error(e.to_string()),
-                    }
+                |e| match e.kind() {
+                    IntErrorKind::PosOverflow | IntErrorKind::NegOverflow => self.mk_parse_error(
+                        "Integer constant is too large.".to_string(),
+                        token,
+                        ParseErrorKind::NotParsable,
+                    ),
+                    _ => self.mk_parse_error(e.to_string(), token, ParseErrorKind::NotParsable),
```

Comment:
Maybe use another ParseErrorKind as `NotParsable` is a bit ambiguous in a context of a language parser. Maybe `NumberFormat`, `NumberNotParsable` would be more appropriate